python/turbine_models/custom_models/llama-benchmark/stateless_llama_benchmark.py

```diff
@@ -0,0 +1,203 @@
+# Copyright 2023 Nod Labs, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import sys
+import numpy as np
+
+from transformers import AutoTokenizer
+from iree import runtime as ireert
+import turbine_models.custom_models.stateless_llama as llama
+
+import argparse
+
+import subprocess
+from collections import namedtuple
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--hf_auth_token", type=str, help="The Hugging Face auth token, required"
+)
+parser.add_argument(
+    "--hf_model_name",
+    type=str,
+    help="HF model name",
+    default="meta-llama/Llama-2-7b-chat-hf",
+)
+parser.add_argument("--external_weight_file", type=str, default="")
+parser.add_argument("--benchmark_vmfb_path", type=str, default="")
+parser.add_argument("--llama_vmfb_path", type=str, default="")
+parser.add_argument(
+    "--steps",
+    type=int,
+    default=10,
+    help="number of times second vicuna is run (# of tokens to benchmark)",
+)
+
+
+def run_benchmark(args):
+    config = ireert.Config("local-task")
+
+    if args.external_weight_file:
+        index = ireert.ParameterIndex()
+        index.load(args.external_weight_file)
+
+    if not args.benchmark_vmfb_path:
+        sys.exit("no benchmark_vmfb_path provided, required for run_benchmark")
+    benchmark_mod = ireert.VmModule.mmap(config.vm_instance, args.benchmark_vmfb_path)
+
+    if not args.llama_vmfb_path:
+        sys.exit("no llama_vmfb_path provided, required for run_benchmark")
+
+    tokenizer = AutoTokenizer.from_pretrained(
+        args.hf_model_name,
+        use_fast=False,
+        use_auth_token=args.hf_auth_token,
+    )
+
+    initial_input = tokenizer(llama.prompt, return_tensors="pt")
+    example_input_id = initial_input.input_ids
+    input = []
+    temp = np.asarray(example_input_id, dtype=None, order="C")
+    input.append(temp)
+    input.append(np.array(args.steps))
+
+    if args.external_weight_file:
+        results = benchmark_module(
+            benchmark_mod,
+            args,
+            "run",
+            input,
+            parameters=f"model={args.external_weight_file}",
+        )
+    else:
+        results = benchmark_module(benchmark_mod, args, "run", input)
+
+    for benchmark_result in results:
+        print(
+            f"benchmark_name: {benchmark_result.benchmark_name}, time: {benchmark_result.time}, cpu_time: {benchmark_result.cpu_time}, iterations: {benchmark_result.iterations}, user_counters: {benchmark_result.user_counters}"
```

Comment:
Might be worth adding a `tok/s` estimate print as well

Reply:
True, that is valuable information. I went ahead and added another benchmarking option for the end user, so that they can get more accurate tok/s by benchmarking only second vicuna (skipping the initialization).